File: src/frontend/src/app/models/lodging.ts
import { Perk } from "./perk";
import { Room } from "./room";
import { RoomType } from "./room_type";
import { User } from "./user";

export class Lodging {
    public roomTypeMaxPrice: number | null = null;
    public roomTypeMinPrice: number | null = null;

    constructor(
        public id:              number,
        public ownerId:         number,
        public type:            any ,
        public name:            string,
        public description:     string,
        public address:         string,
        public emailAddress:    string,
        public perNightPrice:   number | null,
        public fees:            number | null,
        public capacity:        number | null,
        public phoneNumbers:    string[] | null,
        public photos:          string[] | null,
        public perks:           Perk[] | null,
        public rooms:           Room[] | null,
        public roomTypes:       RoomType[] | null,
        public owner:           User | null)
        { }

    public static offersRooms(lodging: Lodging) {
        return this.typeOffersRooms(lodging.type);
    }

    public static typeOffersRooms(lodgingType: number) {
        return lodgingType != LodgingType.Apartment && lodgingType != LodgingType.VacationRental;
    }
}

export enum LodgingType
{
	Apartment,
	GuestHouse,
	Hotel,
	Lodge,
	Motel,
	VacationRental
}

File: src/backend/src/api/Models/Lodging.cs
namespace Restify.API.Models;

public class Lodging
{
	public int		Id { get; set; }
	public int		OwnerId { get; set; }
    public string	Address { get; set; }
    public string	Description { get; set; }
    public LodgingType Type { get; set; }
	public string	Name { get; set; }
	public string	EmailAddress { get; set; }

	public List<LodgingPhoneNumber>	PhoneNumbers { get; }
	public List<LodgingPhoto>			Photos { get; }
	public List<Perk>					Perks { get; }
	public List<Room>					Rooms { get; }
	public List<RoomType>				RoomTypes { get; }
	public Person						Owner { get; }
	
	public Lodging() {}

	public Lodging(List<Room>? rooms, List<RoomType>? roomTypes)
	{
		Rooms = rooms;
		RoomTypes = roomTypes;
	}

	public static bool OffersRooms(Lodging lodging)
	{
        return TypeOffersRooms(lodging.Type);
	}
	
	public static bool TypeOffersRooms(LodgingType type)
	{
        return type is not LodgingType.Apartment and not LodgingType.VacationRental;
	}
}

public enum LodgingType
{
	Apartment,
	GuestHouse,
	Hotel,
	Lodge,
	Motel,
	VacationRental
}
